UnityProject/Assets/Scripts/Items/Wieldable.cs

```diff
@@ -0,0 +1,182 @@
+using Messages.Server;
+using Mirror;
+using UnityEngine;
+
+namespace Items
+{
+	public class Wieldable : NetworkBehaviour, IServerInventoryMove, ICheckedInteractable<HandActivate>
+	{
+		[SerializeField]
+		private int damageUnwielded;
+
+		[SerializeField]
+		private int damageWielded;
+
+		public ItemsSprites Wielded = new ItemsSprites();
+		public ItemsSprites Unwielded = new ItemsSprites();
+
+		[SyncVar(hook = nameof(SyncState))]
+		private bool isWielded;
+
+		private SpriteHandler spriteHandler;
+		private ItemAttributesV2 itemAttributes;
+		private HandsController HandsController;
+
+		public enum hiddenHandValues
+		{
+			bothHands = 0,
+			leftHand = 1,
+			rightHand = 2
+		}
+
+		private void Awake()
+		{
+			itemAttributes = GetComponent<ItemAttributesV2>();
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+		}
+
+		private void OnEnable()
+		{
+			HandsController.OnSwapHand.AddListener(OnSwapHands);
+		}
+
+		private void OnDisable()
+		{
+			HandsController.OnSwapHand.RemoveListener(OnSwapHands);
+		}
+
+		private void OnSwapHands()
+		{
+			if (isWielded)
+			{
+				Chat.AddExamineMsgFromServer(PlayerManager.LocalPlayerScript.gameObject, $"Your other hand is too busy holding {gameObject.ExpensiveName()}!");
+				HandsController.OnSwapHand.RemoveListener(OnSwapHands);
+				HandsController.SwapHand();
+				HandsController.OnSwapHand.AddListener(OnSwapHands);
+			}
+		}
+
+		private void SyncState(bool oldState, bool newState)
+		{
+			isWielded = newState;
+
+			if (isWielded)
+			{
+				itemAttributes.SetSprites(Wielded);
+			}
+			else
+			{
+				itemAttributes.SetSprites(Unwielded);
+			}
+		}
+
+		public void OnInventoryMoveServer(InventoryMove info)
+		{
+			if (info.InventoryMoveType == InventoryMoveType.Remove)
+			{
+				isWielded = false;
+				ItemSlot hiddenHand = DetermineHiddenHand();
+				itemAttributes.ServerHitDamage = damageUnwielded;
+				itemAttributes.SetSprites(Unwielded);
+
+				if (hiddenHand != null)
+				{
+					int hiddenHandSelection = (int) hiddenHandValues.bothHands;
+
+					if (hiddenHand.NamedSlot.GetValueOrDefault(NamedSlot.none) == NamedSlot.leftHand)
+					{
+						hiddenHandSelection = (int) hiddenHandValues.leftHand;
+					}
+					else if (hiddenHand.NamedSlot.GetValueOrDefault(NamedSlot.none) == NamedSlot.rightHand)
+					{
+						hiddenHandSelection = (int) hiddenHandValues.rightHand;
+					}
+
+					HideHand(PlayerManager.LocalPlayerScript.connectionToClient, false, hiddenHandSelection);
+				}
+			}
+
+
+		}
+
+		[TargetRpc]
+		private void HideHand(NetworkConnection target, bool HideState, int HiddenHandSelection)
+		{
+			HandsController.Instance.HideHands(HideState, HiddenHandSelection);
```

Comment:
The only issue with this is that if you have multiple hands, it will hide multiple, 
so you say hide the left hand, it would hide all left hands, 
this is not too bad but , is unexpected behaviour